internal/beater/java_attacher/java_attacher.go

```diff
@@ -361,8 +436,12 @@ func (j *JavaAttacher) attachJVM(ctx context.Context, jvm *jvmDetails) error {
 }
 
 func (j *JavaAttacher) attachJVMCommand(ctx context.Context, jvm *jvmDetails) *exec.Cmd {
+	attacherJar := bundledJavaAttacher
+	if j.tempAttacherDir != "" {
+		attacherJar = tmpJavaAttacherJarPath(j.tempAttacherDir)
+	}
```

Comment:
Yes, I don't like it either, but since this is called from `attachJvm`, which is used within `foreachJVM` that currently requires a specific function structure, I did it this way. I could change that of course, but this is planned to change soon to support a tmp dir per user, where it will make more sense that the more complex state (a map from user to tmp-dir) is maintained by the attacher (you cannot do that from `Run` if you loop over multiple JVMs). 

> Instead of storing tempAttacherDir, we could store a path to the attacher jar, whether it's in a temp directory or not.

But I want to delete the tmp dir and its contents when `Run` exits, so I want a reference to that

Reply:
Changed to tmp dir per user as planned